src/Describer/AnnotationDescriber.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopping\ApiTKDtoMapperBundle\Describer;
+
+use Doctrine\Common\Annotations\Reader;
+use FOS\RestBundle\View\ViewHandlerInterface;
+use Nelmio\ApiDocBundle\Describer\ModelRegistryAwareInterface;
+use Nelmio\ApiDocBundle\Describer\ModelRegistryAwareTrait;
+use Nelmio\ApiDocBundle\Model\Model;
+use Nelmio\ApiDocBundle\OpenApiPhp\Util;
+use Nelmio\ApiDocBundle\RouteDescriber\RouteDescriberInterface;
+use Nelmio\ApiDocBundle\RouteDescriber\RouteDescriberTrait;
+use OpenApi\Annotations as OA;
+use ReflectionAttribute;
+use ReflectionException;
+use ReflectionMethod;
+use ReflectionNamedType;
+use Shopping\ApiTKDtoMapperBundle\Annotation as DtoMapper;
+use Shopping\ApiTKDtoMapperBundle\Service\StringHelper;
+use Symfony\Component\PropertyInfo\Type;
+use Symfony\Component\Routing\Route;
+
+/**
+ * Class AnnotationDescriber.
+ *
+ * Auto generate 200-responses by the annotated dtoMapper.
+ *
+ * The following conditions must match:
+ * * No Response(200) annotation given
+ * * Dto\View annotation with dtoMapper given
+ * * Corresponding dtoMapper has a return typehint
+ *
+ * * Controller action has a return-annotation, which states the return of an array or not (f.e. * @ return Foobar[])
+ */
+class AnnotationDescriber implements ModelRegistryAwareInterface, RouteDescriberInterface
+{
+    use ModelRegistryAwareTrait;
+    use RouteDescriberTrait;
+
+    public function __construct(
+        private Reader $annotationReader,
+        private StringHelper $stringHelper,
+        private ViewHandlerInterface $viewHandler
+    ) {
+    }
+
+    public function describe(OA\OpenApi $api, Route $route, ReflectionMethod $reflectionMethod): void
+    {
+        $annotations = $this->getAnnotations($reflectionMethod);
+        foreach ($this->getOperations($api, $route) as $operation) {
+            foreach ($annotations as $annotation) {
+                $isArray = $this->willReturnArray($reflectionMethod);
+                $type = $this->getDtoByMapper((string) $annotation->getDtoMapper(), $isArray);
+                if (!$type) {
+                    return;
+                }
+
+                // FIXME?
```

Comment:
Is this still an open task to resolve before merging this pull request?

Reply:
No. This is something that can be added at a later point in time via a feature release.

There's no easy way to manipulate the nelmio/api-doc-bundle behaviour of `ModelRegistry` here without overriding the whole registry class. Most of the methods are private so extending it is also not an option.

The best way might be to just contribute upstream 🤷 
Creating a PR against nelmio/api-doc-bundle which allows to customize the model name or circumvent the short name (using an fqcn instead).

This issue has been part of the ModelRegistry since we started using it in 2019, has logging in place and posed no issue in our daily usage, so I'd rather postpone fixing it until there's a need for it.